packages/widgetbook/lib/src/theming/widgetbook_theme.dart

```diff
@@ -0,0 +1,13 @@
+import 'package:flutter/material.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'widgetbook_theme.freezed.dart';
+
+@freezed
+class WidgetbookTheme<CustomTheme> with _$WidgetbookTheme<CustomTheme> {
+  factory WidgetbookTheme({
+    required String name,
+    required IconData icon,
+    required CustomTheme data,
+  }) = _WidgetbookTheme;
```

Comment:
The icon might be a problem for the generator as `IconData` is a flutter dependency